metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py

```diff
@@ -249,14 +249,28 @@ def get_workunits_internal(
 
         with self.report.audit_log_preprocessing_timer:
             # Preprocessing stage that deduplicates the queries using query hash per usage bucket
-            queries_deduped: FileBackedDict[Dict[int, ObservedQuery]]
+            # Using regular dictionary with
+            #   key: usage bucket
+            #   value: File backed dictionary with query hash as key and observed query as value
+            # This structure is chosen in order to maintain order of execution of queries
+
+            queries_deduped: Dict[int, FileBackedDict[ObservedQuery]]
             queries_deduped = self.deduplicate_queries(queries)
-            self.report.num_unique_queries = len(queries_deduped)
+            self.report.num_unique_queries = len(
+                set(
+                    query_hash
+                    for bucket in queries_deduped.values()
+                    for query_hash in bucket
+                )
+            )
 
         with self.report.audit_log_load_timer:
             i = 0
-            for query_instances in queries_deduped.values():
-                for _, query in query_instances.items():
+            for queries_in_bucket in queries_deduped.values():
+                # Assuming that FileBackedDict is an OrderedDict.
+                # If this is not the case, we would need to order it by "last_query_timestamp"
+                # using queries_in_bucket.sql_query_iterator.
```

Comment:
i'm not sure this assumption is valid - might need to write some tests to valid this, but I think sqlite doesn't guarantee anything

Reply:
yeah we will need to tweak the code here https://github.com/datahub-project/datahub/blob/3d5735cbc5899bf294ab3858a78cb9094dea3136/metadata-ingestion/src/datahub/utilities/file_backed_collections.py#L353

Reply:
I'm not superfamilar with file_backed_collections code. I updated the queries_extractor to use order by sql query when reading the results.